frontend/lib/views/photos.dart

```diff
@@ -8,11 +13,63 @@ class PhotosView extends StatefulWidget {
 }
 
 class _PhotosViewState extends State<PhotosView> with AutomaticKeepAliveClientMixin {
+  final PictureService _pictureService = PictureService(pictureRepository: PictureInMemoryRepository());
+  late Future<List<PictureData>> _getUserPicturesFuture;
+  List<PicturesGroupData>? pictureGroups;
+
+  PictureGroupType _currentGroupType = PictureGroupType.day;
+  bool _isGroupTypeAlreadyChanged = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _getUserPicturesFuture = _pictureService.getUserPictures();
+  }
+
   @override
   Widget build(BuildContext context) {
     super.build(context);
-    return const Center(
-      child: Text("Photos"),
+
+    return FutureBuilder<List<PictureData>>(
+      future: _getUserPicturesFuture,
+      builder: (BuildContext context, AsyncSnapshot<List<PictureData>> snapshot) {
+        if (snapshot.hasData) {
+          pictureGroups ??= _pictureService.generatePicturesGroups(PictureGroupType.day, snapshot.data!);
+          
+          return GestureDetector(
+            onScaleUpdate: (ScaleUpdateDetails details) {
+              if (details.scale == 1.0 || _isGroupTypeAlreadyChanged) return;
+              
+              PictureGroupType newGroupType = _currentGroupType;
+
+              if (details.scale < 1.0) {
+                newGroupType = PictureGroupTypeUtils.next(_currentGroupType);
+              }
+              else {
+                newGroupType = PictureGroupTypeUtils.previous(_currentGroupType);
+              }
+              
+              setState(() {
+                _currentGroupType = newGroupType;
+                pictureGroups = _pictureService.generatePicturesGroups(newGroupType, snapshot.data!);
+              });
+
+              _isGroupTypeAlreadyChanged = true;
+            },
+            onScaleEnd: (_) {
+              _isGroupTypeAlreadyChanged = false;
+            },
+            child: PictureList(
+              groupType: _currentGroupType, 
+              data: pictureGroups ?? []
+            ),
+          );
+        } else {
```

Comment:
You did not handle the case where there is an error, showing a circular progress bar is good but you need to do something if an error occurs